v3/src/components/graph/components/graph-drop-hint.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { useLayoutEffect, useRef, useState} from "react"
+import { AxisPlace } from "../models/axis-model"
+
+import "./graph-drop-hint.scss"
+
+interface IProps {
+  hintText: string | undefined
+  targetW: number | undefined
+  targetH: number | undefined
+  place?: string | AxisPlace | undefined
+}
+
+export const GraphDropHint = ({ hintText, targetW, targetH, place }: IProps) => {
+  const hintDiv = useRef<HTMLDivElement>(null)
+  const [hintFoundW, setHintFoundW] = useState(0);
+  const [hintFoundH, setHintFoundH] = useState(0);
+
+  useLayoutEffect(() => {
+    if (hintDiv.current){
+      setHintFoundW(hintDiv.current?.offsetWidth)
+      setHintFoundH(hintDiv.current?.offsetHeight)
+    }
+  }, [hintDiv]);
+
+  function getLeftVal(wrapW: number, hintW: number){
+    return (wrapW * .5) - (hintW * .25)
```

Comment:
This calculation - it appears to work at the moment, but I am suspicious.  Shouldn't it be this: https://codepen.io/joe_bacal/pen/xxjvpVK

Reply:
now entirely using mouse position